Add publication approved overlay notification.

ankitrox opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #8843

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

nfmohit commented on assets/js/modules/reader-revenue-manager/components/PublicationApprovedOverlayNotification.js:
```diff
@@ -0,0 +1,141 @@
+					{ __( 'Maybe later', 'google-site-kit' ) }
+				</Button>
+
+				<Link
+					className="mdc-button mdc-button--raised mdc-ripple-upgraded"
+					external
+					href={ serviceURL }
+					inverse
+					linkButton
+				>
+					{ __( 'Enable features', 'google-site-kit' ) }
+				</Link>
```
Using the `Button` component here would be wiser IMO, as this is technically and semantically a button. The `Button` component does support links.

```suggestion
				<Button
					href={ serviceURL }
					trailingIcon={ <ExternalIcon width={ 10 } height={ 10 } /> }
					target="_blank"
				>
					{ __( 'Enable features', 'google-site-kit' ) }
				</Button>
```

nfmohit commented on assets/js/modules/reader-revenue-manager/components/PublicationApprovedOverlayNotification.test.js:
```diff
@@ -0,0 +1,135 @@
+describe( 'PublicationApprovedOverlayNotification', () => {
+	let registry;
+
+	beforeAll( () => {
+		registry = createTestRegistry();
+	} );
```
We usually reset the registry after each test case (i.e. in `beforeEach()`). Is there a reason we're doing it here once?

nfmohit commented on assets/js/modules/reader-revenue-manager/components/PublicationApprovedOverlayNotification.test.js:
```diff
@@ -0,0 +1,135 @@
+
+		const { container, waitForRegistry } = render(
+			<ViewContextProvider
+				value={ VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY }
```
We should change this to be the main dashboard as the component would not render in the view-only dashboard even when it is not dismissed.

```suggestion
				value={ VIEW_CONTEXT_MAIN_DASHBOARD }
```

nfmohit commented on assets/sass/components/overlay-notification/_googlesitekit-overlay-notification.scss:
```diff
@@ -98,4 +98,18 @@
 	}
+
+	&.googlesitekit--overlay-notification--reader-revenue-manager {
+		max-width: 348px;
```
This max-width should be applied only on viewport `$bp-tablet` and up.

nfmohit commented on assets/sass/components/overlay-notification/_googlesitekit-overlay-notification.scss:
```diff
@@ -98,4 +98,18 @@
+		max-width: 348px;
+
+		.mdc-button--raised {
+			@extend .mdc-button--raised;
```
This looks redundant. Do you think this is necessary?

ankitrox replied:
Removed this one.

nfmohit commented on assets/sass/components/overlay-notification/_googlesitekit-overlay-notification.scss:
```diff
@@ -98,4 +98,18 @@
+		.mdc-button--raised {
+			@extend .mdc-button--raised;
+			font-weight: $fw-medium;
+			padding: 10px $grid-gap-phone;
```
This won't be needed if the `Button` component is used.

nfmohit commented on assets/sass/components/overlay-notification/_googlesitekit-overlay-notification.scss:
```diff
@@ -98,4 +98,18 @@
 		}
 	}
+
+	&.googlesitekit--overlay-notification--reader-revenue-manager {
```
I would suggest moving the styles for this component to its own stylesheet, e.g. `assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-overlay-notification.scss`.

nfmohit commented on assets/svg/graphics/reader-revenue-manager-introductory-graphic-desktop.svg:
It looks like there is an issue with the exported graphic. There is a shadow around the subject which does not appear in the Figma designs.

ankitrox replied:
I've asked @sigal-teller to kindly look into the issue.

nfmohit commented on assets/svg/graphics/reader-revenue-manager-introductory-graphic-mobile.svg:
It looks like there is an issue with the exported graphic. There is a shadow around the subject which does not appear in the Figma designs.

ankitrox replied:
I've asked @sigal-teller to kindly look into the issue.

nfmohit reviewed (changes requested):
Thank you for your nice work on this, @ankitrox . Please take a look at my comments and feel free to reach out if you have any questions, thanks!

ankitrox commented on assets/js/modules/reader-revenue-manager/components/PublicationApprovedOverlayNotification.js:
I agree, it makes more sense.

ankitrox commented:
Thanks @nfmohit for your review.

I've addressed the feedback added and sending this your way for another review.

nfmohit commented on assets/js/modules/reader-revenue-manager/components/dashboard/PublicationApprovedOverlayNotification.js:
```diff
@@ -0,0 +1,139 @@
+			RRM_PUBLICATION_APPROVED_OVERLAY_NOTIFICATION
+		);
+	};
+
+	if ( isDismissing ) {
+		return null;
+	}
```
I feel this is redundant and may even have unexpected side effects, for example, if the dismissal request fails for any reason (e.g. a network error), we may see a flash of the notification disappearing and reappearing. We also do not have this check for other overlay notifications. What do you think?

```suggestion
```

nfmohit commented on assets/js/modules/reader-revenue-manager/components/dashboard/PublicationApprovedOverlayNotification.js:
```diff
@@ -0,0 +1,139 @@
+					{ __( 'Maybe later', 'google-site-kit' ) }
+				</Button>
+
+				<Button
```
This button should also be disabled when `isDismissing` is truthy.

nfmohit commented on assets/js/modules/reader-revenue-manager/components/dashboard/PublicationApprovedOverlayNotification.js:
```diff
@@ -0,0 +1,139 @@
+
+				<Button
+					href={ serviceURL }
+					trailingIcon={ <ExternalIcon width={ 14 } height={ 14 } /> }
```
According to the Figma designs, I believe the dimensions here should be `13`.

```suggestion
					trailingIcon={ <ExternalIcon width={ 13 } height={ 13 } /> }
```

nfmohit commented on assets/js/modules/reader-revenue-manager/components/dashboard/PublicationApprovedOverlayNotification.test.js:
```diff
@@ -0,0 +1,130 @@
+import { render } from '../../../../../../tests/js/test-utils';
+import PublicationApprovedOverlayNotification, {
+	RRM_PUBLICATION_APPROVED_OVERLAY_NOTIFICATION,
+} from '../dashboard/PublicationApprovedOverlayNotification';
```
```suggestion
} from './PublicationApprovedOverlayNotification';
```

nfmohit commented on assets/js/modules/reader-revenue-manager/components/dashboard/index.js:
```diff
@@ -0,0 +1,19 @@
+/**
+ * RRM Dashboard components.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
```
```suggestion
 * RRM dashboard components.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
```

nfmohit commented on assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-overlay-notification.scss:
```diff
@@ -0,0 +1,29 @@
+ * limitations under the License.
+ */
+
+.googlesitekit-overlay-notification {
+	&.googlesitekit-reader-revenue-manager-publication-approved-notification {
```
1. We usually have most of our styles as descendants of the `.googlesitekit-plugin` class.
2. We can reduce the specificity of this selector, i.e. we don't need the `.googlesitekit-overlay-notification` sibling selector.
3. We usually follow a convention where we add an empty line before every selector.

Please see suggestion:

```suggestion
.googlesitekit-plugin {

	.googlesitekit-reader-revenue-manager-publication-approved-notification {

```

nfmohit commented on assets/sass/components/reader-revenue-manager/_index.scss:
```diff
@@ -0,0 +1,17 @@
+/**
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
```
Extreme nit, but while we're at it:

```suggestion
/**
 * Reader Revenue Manager styles.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
```

nfmohit commented on assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-overlay-notification.scss:
```diff
@@ -0,0 +1,29 @@
+
+.googlesitekit-overlay-notification {
+	&.googlesitekit-reader-revenue-manager-publication-approved-notification {
+		.mdc-button--raised {
+			font-weight: 500;
+		}
```
The button text and icon should have a gap of 6px, and we should use appropriate variable for the font weight:

```suggestion
		.mdc-button--raised {
			gap: 0 6px;
			font-weight: 500;
		}
```

nfmohit commented on tests/backstop/reference/google-site-kit_Modules_ReaderRevenueManager_Components_PublicationApprovedOverlayNotification_0_document_0_small.png:
While the graphics have been updated, for some reason, the VRT references still have the shadow around the graphic. I think this is why VRT is failing. Could you take a look?

nfmohit reviewed (changes requested):
Thank you for addressing my feedback, @ankitrox. I apologise for the back and forth, but I have left another set of feedback on the follow-up, which I hope should be quite straightforward. Please let me know if you have any questions, thank you!

ankitrox commented:
Thanks for reviewing this @nfmohit 

I've addressed the comments added by you and sending your way for another review. I've also updated the VRT reference images in the PR.

nfmohit commented on assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-overlay-notification.scss:
```diff
@@ -0,0 +1,32 @@
+
+.googlesitekit-plugin {
+
+	&.googlesitekit-reader-revenue-manager-publication-approved-notification {
```
The specificity is incorrect here, causing the styles to be broken.

```suggestion
	.googlesitekit-reader-revenue-manager-publication-approved-notification {
```

nfmohit reviewed (changes requested):
Thank you for addressing my feedback, @ankitrox.

It looks like the CSS caused the styles to break which I've highlighted below. It looks like a small fix, so I'll try and see if that resolves the issue.

nfmohit reviewed (approved):
LGTM ✅
